IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/ConditionalWait.cs

```diff
@@ -0,0 +1,113 @@
+﻿using OpenQA.Selenium;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IntelliTect.TestTools.Selenate
+{
+    public class ConditionalWait
```

Comment:
You may also consider making this a static class. Right now most of the code creates a new instance, but this object does not hold any data and merely calls pure methods.

Reply:
So you might be mixing some concepts, instance vs static methods have no real affect when we are discussing async methods. So no, it will not cause it to get hung up.